### ISSUE TITLE
Fatal Python Error While Fitting

### ISSUE THREAD

**Issue by cyntqliu** [aigamedev/scikit-neuralnetwork#214]
The following code, part of a program for playing tic-tac-toe using only a critic in an actor-critic model, produces a fatal python error while fitting:

def main():

```
#Initialize the board - the state
state = np.ndarray(shape=(3,3), dtype=int)
actions = [[0,0], [0,1], [0,2], [1,0], [1,1], [1,2], [2,0], [2,1], [2,2]] #also spaces
moveNum = 1

#Variables
gamma = .3
tau = .02

critic = Regressor(layers=[Layer("Rectifier", name="layer1", units=11, pieces=2), #9 squares, 1 action, 1 bias
                           Layer("Rectifier", name="layer2", units=11, pieces=2),
                           Layer("Rectifier", name="layer3", units=11, pieces=2),
                           Layer("Softmax")], learning_rate=0.005)

#Randomly initialize the critic
statesAndActs = []; rewards = []
for i in range(3):
    sample_st = np.ndarray(shape=(3,3), dtype=int)
    for j in range(9):
        sample_st[math.floor(j/3),j%3] = int(random.randint(-1,1))

    if i == 0:
        print sample_st

    #random action, random reward
    act = random.randint(0,8)
    rew = random.randint(-1,1)
    rewards.append(rew)

    stateAndAct = []
    for k in range(9):
        stateAndAct.append(sample_st[int(k/3),k%3])

    stateAndAct.append(act)
    if i == 0:
        print stateAndAct
    statesAndActs.append(stateAndAct)

sA = np.array(statesAndActs); r = np.array(rewards)
print sA, r
critic.fit(sA,r)
```

The issue still occurs even if I am training the critic on 1 input, or 500. Thank you!

**Comment by alexjc** [comment]
If the fatal error is `RuntimeError` then follow the instructions on screen to adjust learning rate. Otherwise, can you paste the error?

**Comment by cyntqliu** [comment]
"Fatal Python error: deallocating None

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information."

I already contacted WING IDE support, and they suggested that I run the program outside of WING to narrow down possibilities for the problem's source. I ran this segment in command line and the program did not terminate for an unusually long amount of time (an hour), which leads me to suspect that the problem is with sknn.

**Comment by alexjc** [comment]
If the program doesn't terminate you should add more logging in there to see what it's doing and why. There are bugs in sknn and the libraries it uses, but there's not enough here to tell exactly where. If you have more information, please re-open and attach what you have!
